drivers/input/input_sbus.c

```diff
@@ -0,0 +1,370 @@
+	struct input_sbus_data *const data = dev->data;
+	int i, ret;
+
+	uart_irq_rx_disable(config->uart_dev);
+	uart_irq_tx_disable(config->uart_dev);
+
+	LOG_DBG("Initializing SBUS driver");
+
+	for (i = 0; i < SBUS_CHANNEL_COUNT; i++) {
+		data->last_reported_value[i] = 0;
+		data->channel_mapping[i] = -1;
+	}
+
+	data->xfer_bytes = 0;
+	data->in_sync = false;
+	data->partial_sync = false;
+	data->last_rx_time = 0;
+
+	for (i = 0; i < config->num_channels; i++) {
+		data->channel_mapping[config->channel_info[i].sbus_channel - 1] = i;
+	}
+
+	ret = uart_configure(config->uart_dev, &uart_cfg_sbus);
+	if (ret < 0) {
+		LOG_ERR("Unable to configure UART port: %d", ret);
+		return ret;
+	}
+
+	ret = uart_irq_callback_user_data_set(config->uart_dev, config->cb, (void *)dev);
+	if (ret < 0) {
+		if (ret == -ENOTSUP) {
+			LOG_ERR("Interrupt-driven UART API support not enabled");
+		} else if (ret == -ENOSYS) {
+			LOG_ERR("UART device does not support interrupt-driven API");
+		} else {
+			LOG_ERR("Error setting UART callback: %d", ret);
+		}
+		return ret;
+	}
+
+	uart_irq_rx_enable(config->uart_dev);
+
+	/* Initialize semaphore used by thread to report input */
+	k_sem_init(&data->report_lock, 0, 1);
+
+	k_thread_create(&data->thread, data->thread_stack,
+			K_KERNEL_STACK_SIZEOF(data->thread_stack),
+			(k_thread_entry_t)input_sbus_input_report_thread, (void *)dev, NULL, NULL,
+			CONFIG_INPUT_SBUS_THREAD_PRIORITY, 0, K_NO_WAIT);
+
+	k_thread_name_set(&data->thread, dev->name);
+
+	return ret;
+}
+
+#define INPUT_CHANNEL_CHECK(input_channel_id)                                                      \
+	BUILD_ASSERT(IN_RANGE(DT_PROP(input_channel_id, channel), 1, 16),                          \
+		     "invalid channel number");                                                    \
+	BUILD_ASSERT(DT_PROP(input_channel_id, type) == INPUT_EV_ABS ||                            \
+			     DT_PROP(input_channel_id, type) == INPUT_EV_KEY ||                    \
+			     DT_PROP(input_channel_id, type) == INPUT_EV_MSC,                      \
+		     "invalid channel type");
+
+#define INPUT_INFO(input_channel_id)                                                               \
+	{                                                                                          \
+		.sbus_channel = DT_PROP(input_channel_id, channel),                                \
+		.type = DT_PROP(input_channel_id, type),                                           \
+		.zephyr_code = DT_PROP(input_channel_id, zephyr_code),                             \
+	},
+
+#define INPUT_SBUS_INIT(n)                                                                         \
+                                                                                                   \
+	static const struct sbus_input_channel input_##id[] = {                                    \
+		DT_INST_FOREACH_CHILD(n, INPUT_INFO)};                                             \
```

Comment:
```c
static const struct sbus_input_channel input_##id[] = { \      
        DT_INST_FOREACH_CHILD(n, INPUT_INFO)            \
};  
```

Reply:
I've renamed it to `SBUS_INPUT_CHANNEL_INITIALIZER` unless you meant something else here.

Reply:
I mean: put the `};` on a new line

Reply:
Fixed it manually, it seems that clang-format is enforcing this to go on a single line.

Reply:
thanks, you can add some `clang-format off` `clang-format on` marker around it if you care about running clang-format on the rest of the file, I don't mind either